corehq/apps/reports/generic.py

```diff
@@ -223,20 +227,24 @@ class FakeHttpRequest(object):
             datespan = None
             can_access_all_locations = None
 
+        if 'startdate' and 'enddate' in state['request_params']:
```

Comment:
not impossible! there's reports with just a startdate but they don't need a datespan object if that's case (which is what's being re-made here since datespan objects are not JSON serializable)

Reply:
thanks for catching that Norman! I said this in another comment but I'm realizing that I may have pulled the trigger on this PR a little too early. Do you suggest I make some sort of amendment PR to address some of these issues or would it be better to revert this PR and continue working on that?

Reply:
Should this PR be reverted? This seems like a bug.

Reply:
@millerdev I think that would be best since there's other small issues (not bugs like this) that came up after I merged it in too early I'd like to address